include/podio/EDMDefinitionRegistry.h

```diff
@@ -0,0 +1,62 @@
+#ifndef PODIO_EDMDEFINITIONREGISTRY_H
+#define PODIO_EDMDEFINITIONREGISTRY_H
+
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+namespace podio {
+class EDMDefinitionRegistry {
```

Comment:
The registry should be a "DatatypeRegistry" which deals with knowing about definitions and how to instantiate classes after just passing a name to it (needed for all non-ROOT cases). I would not like having multiple registries around.

Reply:
Yes, I agree the name is not the best here. Having one single registry would be nice, but it potentially becomes rather large, depending on what we want to do with it in the end:
- Getting the JSON encoded yaml file
- Creating collection buffers from a type name (and potentially a version)
- Getting the names of the relations of a given data type (would be nice to address #169)
- Do we also want schema evolution hooked into this? I.e. get an evolution function given a type name, and a buffer version (and potentially a backend)?

I think the first three can easily go into this one, I am not entirely sure about the schema evolution yet, but maybe we can see how that works out when we get there.

Reply:
I think we should go for the first 3 here you list.
For the schema evolution let's discuss offline.

Reply:
I think that makes sense. I would still defer the other points to separate PRs to not let this grow too large. The main thing to address in this PR is the naming of the Registry, which should now be general enough to allow for adding the other two functionalities as well without causing confusion.